pkg/storage/iterator_test.go

```diff
@@ -639,6 +643,130 @@ func TestPartitionOverlappingchunks(t *testing.T) {
 	}
 }
 
+func TestBuildHeapIterator(t *testing.T) {
+	var (
+		firstChunk = newLazyChunk(logproto.Stream{
+			Labels: "{foo=\"bar\"}",
+			Entries: []logproto.Entry{
+				{
+					Timestamp: from,
+					Line:      "1",
+				},
+				{
+					Timestamp: from.Add(time.Millisecond),
+					Line:      "2",
+				},
+				{
+					Timestamp: from.Add(2 * time.Millisecond),
+					Line:      "3",
+				},
+			},
+		})
+		secondChunk = newLazyInvalidChunk(logproto.Stream{
+			Labels: "{foo=\"bar\"}",
+			Entries: []logproto.Entry{
+				{
+					Timestamp: from.Add(3 * time.Millisecond),
+					Line:      "4",
+				},
+				{
+					Timestamp: from.Add(4 * time.Millisecond),
+					Line:      "5",
+				},
+			},
+		})
+		thirdChunk = newLazyChunk(logproto.Stream{
+			Labels: "{foo=\"bar\"}",
+			Entries: []logproto.Entry{
+				{
+					Timestamp: from.Add(5 * time.Millisecond),
+					Line:      "6",
+				},
+			},
+		})
+	)
+
+	for i, tc := range []struct {
+		input    [][]*chunkenc.LazyChunk
+		expected []*logproto.Stream
+	}{
+		{
+			[][]*chunkenc.LazyChunk{
+				{firstChunk},
+				{thirdChunk},
+			},
+			[]*logproto.Stream{
+				{
+					Labels: "{foo=\"bar\"}",
+					Entries: []logproto.Entry{
+						{
+							Timestamp: from,
+							Line:      "1",
+						},
+						{
+							Timestamp: from.Add(time.Millisecond),
+							Line:      "2",
+						},
+						{
+							Timestamp: from.Add(2 * time.Millisecond),
+							Line:      "3",
+						},
+						{
+							Timestamp: from.Add(5 * time.Millisecond),
+							Line:      "6",
+						},
+					},
+				},
+			},
+		},
+		{
+			[][]*chunkenc.LazyChunk{
+				{secondChunk},
+				{firstChunk, thirdChunk},
+			},
+			[]*logproto.Stream{
+				{
+					Labels: "{foo=\"bar\"}",
+					Entries: []logproto.Entry{
+						{
+							Timestamp: from,
+							Line:      "1",
+						},
+						{
+							Timestamp: from.Add(time.Millisecond),
+							Line:      "2",
+						},
+						{
+							Timestamp: from.Add(2 * time.Millisecond),
+							Line:      "3",
+						},
+						{
+							Timestamp: from.Add(5 * time.Millisecond),
+							Line:      "6",
+						},
+					},
+				},
+			},
+		},
+	} {
+		t.Run(fmt.Sprintf("%d", i), func(t *testing.T) {
+			ctx = user.InjectOrgID(context.Background(), "test-user")
```

Comment:
See I think this test is sufficient. Remove any changes from the other tests in `pkg/storage/store_test.go ` and other changes for that tests, let's keep only this one. And rename the chunk variable to `invalidChunk`.